src/java/org/apache/cassandra/index/SecondaryIndexManager.java

```diff
@@ -1329,15 +1329,18 @@ private Index unregisterIndex(String name)
         if (removed != null)
```

Comment:
An alternative approach would be:
- add `public void unregisterIndex(Index removed, Object groupKey)` to `IndexRegistry`, similar to existing `IndexRegistry#registerIndex(Index index, Object groupKey, Supplier<Index.Group> groupSupplier)`. 
- add `public void unregister(IndexRegistry registry)` to `Index`, similar to existing `Index#registry`. Each index implementation will call index registry to unregister itself with proper `groupKey`